cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerParcelManagementService.java

```diff
@@ -0,0 +1,115 @@
+import com.sequenceiq.cloudbreak.cm.model.ParcelResource;
+import com.sequenceiq.cloudbreak.cm.polling.ClouderaManagerPollingServiceProvider;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.polling.PollingResult;
+import com.sequenceiq.cloudbreak.service.CloudbreakException;
+
+@Service
+class ClouderaManagerParcelManagementService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClouderaManagerParcelManagementService.class);
+
+    @Inject
+    private ClouderaManagerPollingServiceProvider clouderaManagerPollingServiceProvider;
+
+    void setParcelRepos(Set<ClouderaManagerProduct> products, ClouderaManagerResourceApi clouderaManagerResourceApi) throws ApiException {
+        Set<String> stackProductParcels = products.stream()
+                .map(ClouderaManagerProduct::getParcel)
+                .collect(Collectors.toSet());
+        LOGGER.info("Setting parcel repo to {}", stackProductParcels);
+        ApiConfigList apiConfigList = new ApiConfigList()
+                .addItemsItem(new ApiConfig()
+                        .name("remote_parcel_repo_urls")
+                        .value(String.join(",", stackProductParcels)));
+        clouderaManagerResourceApi.updateConfig("Updated configurations.", apiConfigList);
+    }
+
+    void refreshParcelRepos(ClouderaManagerResourceApi clouderaManagerResourceApi, Stack stack, ApiClient apiClient) {
+        try {
+            LOGGER.info("Refreshing parcel repos.");
+            ApiCommand apiCommand = clouderaManagerResourceApi.refreshParcelRepos();
+            clouderaManagerPollingServiceProvider.startPollingCmParcelRepositoryRefresh(stack, apiClient, apiCommand.getId());
+        } catch (ApiException e) {
+            LOGGER.info("Unable to refresh parcel repo", e);
+            throw new ClouderaManagerOperationFailedException(e.getMessage(), e);
+        }
+    }
+
+    void downloadParcels(Set<ClouderaManagerProduct> products, ParcelResourceApi parcelResourceApi, Stack stack, ApiClient apiClient)
+            throws ApiException, CloudbreakException {
+        for (ClouderaManagerProduct product : products) {
+            LOGGER.info("Downloading {} parcel.", product.getName());
+            ApiCommand apiCommand = parcelResourceApi.startDownloadCommand(stack.getName(), product.getName(), product.getVersion());
+            PollingResult pollingResult = clouderaManagerPollingServiceProvider.startPollingCdpRuntimeParcelDownload(
+                    stack, apiClient, apiCommand.getId(), new ParcelResource(stack.getName(), product.getName(), product.getVersion()));
+            if (isExited(pollingResult)) {
+                throw new CancellationException("Cluster was terminated while waiting for CDP Runtime Parcel to be downloaded");
+            } else if (isTimeout(pollingResult)) {
+                throw new CloudbreakException("Timeout during the updated CDP Runtime Parcel download.");
+            }
```

Comment:
this part is duplicated a few times, message maybe different a bit, but that could be a parameter